src/file/watcher.go

```diff
@@ -67,9 +65,15 @@ func NewWatcher(e *env.Env, configPath string, checksums map[string]string) (*Wa
 	}
 	fsWatcher.AddFilterHook(hook)
 
-	if err := fsWatcher.AddRecursive(e.Directory); err != nil {
+	if err := fsWatcher.Add(e.Directory); err != nil {
 		return nil, fmt.Errorf("Could not watch directory: %s", err)
 	}
+	for _, folder := range assetLocations {
+		path := filepath.Join(e.Directory, folder)
+		if err := fsWatcher.Add(path); err != nil && !os.IsNotExist(err) {
```

Comment:
Should this be `IsNotExist(path)`?

Reply:
no, this is a helper that checks if the error is an error about it not existing https://golang.org/pkg/os/#IsNotExist The type system would not allow me to pass the wrong thing to it.